messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -262,4 +262,18 @@ public MessageCheckpoint getMessageCheckpoint() {
   public MessageCheckpoint buildStartPartitionCheckpoints() {
     return null;
   }
+
+  @Override
+  public Long getReaderBackLog() throws IOException {
+    Long pendingSize = 0l;
+    if (reader == cReader) {
```

Comment:
if reader == creader, pending collector data will be added, else pending local stream data will be added. That was the idea behind this piece of code, your comment is indicating that my understanding is wrong, I will correct on this.